staff/acct/signat

```diff
@@ -1,104 +1,173 @@
-    except (ldap.INVALID_DN_SYNTAX, ldap.NO_SUCH_OBJECT):
-        raise ValueError(("Invalid DN: " + ldap_basedn))
-    if entry:
-        try:
-            return int(entry[0][1][ldap_attribute][0])
-        except KeyError:
-            raise KeyError(("Account does not have a CalNet UID: " + username))
-    else:
-        raise ValueError(("Account not found in LDAP: " + username))
+
+from tabulate import tabulate
+
+import ocflib.account.search as search
+import ocflib.constants as constants
+import ocflib.infra.ldap as ldap
+import ocflib.ucb.directory as directory
+from ocflib.ucb.groups import *
+from ocflib.misc.shell import red
+
+
+def print_table(title, columns, headers):
+    """Pretty print a table of groups or signatories."""
+    bar = '=' * len(title)
+    print(title)
+    print(bar)
+    print(tabulate(columns, headers=headers))
+
+
+def show_groups_by_signat_username(username):
+    """Show group signatory status of an OCF user."""
+    attrs = search.user_attrs(username)
+    if attrs is None:
+        print(red("Username not found."))
+        return
+    elif 'calnetUid' not in attrs:
+        print(red("CalNet UID not found."))
+        return
+
+    show_groups_by_student_signat(int(attrs['calnetUid'][0]))
+
+def show_signatories_for_group(oid, uids=None):
+    """Show all signatories for a group."""
+    if uids is None:
+        signatories = signatories_for_group(oid)
+
+    if len(signatories) == 0:
+        print(red("No signatories found."))
+        return
+
+    name = group_by_oid(oid)['name']
+    title = "{} ({})".format(name, oid)
+    columns = [(attrs['name'], uid) for (uid, attrs) in signatories.items()]
+    headers = ('Signatory', 'UID')
+
+    print_table(title, columns, headers)
+
+
+def show_groups_by_student_signat(uid, groups=None):
+    """Show signatory status of a person on CalLink."""
+    if groups is None:
+        groups = groups_by_student_signat(uid)
+
+    if len(groups) == 0:
+        print(red("No active student groups found."))
+        return
+
+    name = directory.name_by_calnet_uid(uid)
+    title = "{} ({})".format(name, uid)
+    columns = [(attrs['name'], uid) for (uid, attrs) in groups.items()]
+    headers = ('Group', 'OID')
+
+    print_table(title, columns, headers)
+
+
+def show_signatories_by_group_name(name):
+    """Show signatories of groups matching a name."""
+    groups = list_groups(name=name)
+
+    print("Searching for groups... ", end="", flush=True)
+
+    if len(groups) == 0:
+        print(red("No student groups found."))
+        return
+
+    print("Found {} entries.".format(len(groups)))
+
+    print("Searching for signatories...")
+
+    found = False
+
+    for (oid, attrs) in groups.items():
+        signatories = signatories_for_group(oid)
+        if signatories:
+            found = True
+            show_signatories_for_group(oid)
+            print()
+
+    if not found:
+        print(red("No groups had signatories."))
```

Comment:
I think you can probably rewrite this as:

```python
for oid, attrs in groups.items():
    signatories = signatories_for_group(oid)
    if signatories:
        show_signatories_for_group(oid)
        break
else:
    print(red("No groups had signatories."))
```

Also, maybe pass the signatories in to that function so we don't look them up twice?